### PR TITLE
addressed comments and removed vulnerability

### LINKED ISSUE #104: Add User Agent Header
Add the user-agent request header to every call in this sdk.

Example:

headers: {
   Authorization: `Bearer ${ApiKey}`,
   "User-Agent": `novu/${sdk}@${version}`,
},

Why? (Context)

Having request headers in the SDKs will help us in monitoring which SDK is mostly used by the community and customers. It will help us make the decisions in the future on which SDKs to officially support in the future.

### PR DESCRIPTION
**Pull request by VishalMCF**
closes #104 

Summary:-
1. Added loadSdkVersionFromPom() which reads the POM file from the location inside the jar where client has installed the Java-SDK dependency.
2. Added dependency for MavenXpp3Reader which enables parsing of maven pom files.

### CODE REVIEW

**Comment by VishalMCF** [comment]
@mayorJAY Please review the changes which I made as per your suggestions.

**Comment by mayorJAY** [comment]
Looks great, just a couple of observations @VishalMCF

File: src/main/java/co/novu/common/rest/RestHandler.java [hunk 1]
```diff
@@ -61,4 +68,16 @@ public <T, R> R extractResponse(Response<T> response, R body) throws NovuNetwork
...
     }
+
+    private String loadSdkVersionFromPom(){
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+            Model model = reader.read(
+                new InputStreamReader(Objects.requireNonNull(this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml"))));
+            return model.getVersion();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/rest/RestHandler.java:79]: Annotate this class with Slf4j and log the exception

File: src/main/java/co/novu/common/rest/RestHandler.java [hunk 2]
```diff
@@ -30,6 +36,7 @@ public Retrofit buildRetrofit() {
                     Request request = chain.request()
                             .newBuilder()
                             .addHeader("Authorization", "ApiKey " + novuConfig.getApiKey())
+                            .addHeader("User-Agent", "novu/JAVA" + "@" + loadSdkVersionFromPom())
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/rest/RestHandler.java:39]:
```suggestion
                            .addHeader("User-Agent", "novu/Java@" + loadSdkVersionFromPom())
```

**Comment by VishalMCF** [comment]
@mayorJAY On it. Will update soon

File: src/main/java/co/novu/common/rest/RestHandler.java (hunk 2 above)
Comment (VishalMCF) [reply src/main/java/co/novu/common/rest/RestHandler.java]: Added the suggested changes

File: src/main/java/co/novu/common/rest/RestHandler.java (hunk 1 above)
Comment (VishalMCF) [reply src/main/java/co/novu/common/rest/RestHandler.java]: Added the suggested changes

File: src/main/java/co/novu/common/rest/RestHandler.java [hunk 3]
```diff
@@ -30,6 +39,7 @@ public Retrofit buildRetrofit() {
                     Request request = chain.request()
                             .newBuilder()
                             .addHeader("Authorization", "ApiKey " + novuConfig.getApiKey())
+                            .addHeader("User-Agent", "novu/JAVA@" + loadSdkVersionFromPom())
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/rest/RestHandler.java:42]:
```suggestion
                            .addHeader("User-Agent", "novu/Java@" + loadSdkVersionFromPom())
```

**Review by mayorJAY — APPROVED** [review]
LGTM 🚀

File: src/main/java/co/novu/common/rest/RestHandler.java [hunk 4]
```diff
@@ -61,4 +71,16 @@ public <T, R> R extractResponse(Response<T> response, R body) throws NovuNetwork
             throw new NovuNetworkException(response.errorBody() != null ? response.errorBody().string() : "Error connecting to Novu API");
         }
     }
+
+    private String loadSdkVersionFromPom() {
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+            Model model = reader.read(
+                new InputStreamReader(Objects.requireNonNull(this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml"))));
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/rest/RestHandler.java:79]:
```suggestion
            InputStream inputStream = this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml");
            if (inputStream == null) {
                return "";
            }
            Model model = reader.read(new InputStreamReader(inputStream));
```

File: src/main/java/co/novu/common/rest/RestHandler.java [hunk 5]
```diff
@@ -4,16 +4,25 @@
...
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;
 
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+
+import java.io.FileReader;
 import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Objects;
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/rest/RestHandler.java:22]:
```suggestion
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
```

File: src/main/java/co/novu/common/rest/RestHandler.java [hunk 6]
```diff
@@ -61,4 +70,16 @@ public <T, R> R extractResponse(Response<T> response, R body) throws NovuNetwork
             throw new NovuNetworkException(response.errorBody() != null ? response.errorBody().string() : "Error connecting to Novu API");
         }
     }
+
+    private String loadSdkVersionFromPom() {
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+            Model model = reader.read(
+                new InputStreamReader(Objects.requireNonNull(this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml"))));
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/rest/RestHandler.java:78]:
```suggestion
            InputStream inputStream = this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml");
            if (inputStream == null) {
                return "";
            }
            Model model = reader.read(new InputStreamReader(inputStream));
```

File: src/main/java/co/novu/common/rest/RestHandler.java [hunk 7]
```diff
@@ -61,4 +70,19 @@ public <T, R> R extractResponse(Response<T> response, R body) throws NovuNetwork
...
     }
+
+    private String loadSdkVersionFromPom() {
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+            InputStream inputStream = this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml");
+            if (inputStream == null) {
+                return "";
+            }
+            Model model = reader.read(new InputStreamReader(inputStream));
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/rest/RestHandler.java:81]:
```suggestion
            InputStream inputStream = this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml");
            if (inputStream == null) {
                return "";
            }
            MavenXpp3Reader reader = new MavenXpp3Reader();
            Model model = reader.read(new InputStreamReader(inputStream));
```

**Review by mayorJAY — APPROVED** [review]

**Comment by VishalMCF** [comment]
@Cliftonz Can this be merged now?

**Pull request merged**